CMakeLists.txt

```diff
@@ -48,6 +49,10 @@ option(BUILD_SHARED_LIBS "Build project as shared libraries." ON)
 option(BUILD_TESTING "Build the testing tree." OFF)
 option(BUILD_HEADLESS "Build only headless components: server and AI." OFF)
 
+# The pdb file produced by the Parsers project is larger than the memory limit of MSVC
+# If you want to debug with Visual Studio, please use this option instead of the default debug configuration
+option(MSVC_DEBUG_INFO "When compile with MSVC, create pdb files for all the subproject except Parsers." OFF)
```

Comment:
Fix this properly. Tune the specific target compile flags instead of sprinkling nonstandard build configurations accross the script.

Reply:
It is surprisingly non-trivial to tune the default flag generated by CMake for Visual Studio project, I guess I can still find a way to do it if it is better.

Actually, I am following the way of the current msvc project, there is only one "Release" configuration which produce the debug files automatically. Well I can do this with CMake as well without the option.

Reply:
This should be doable via
`get_target_property` with `COMPILE_OPTIONS` option
`list` with `REMOVE_ITEM` option pruning `/Zi` from the list
`set_target_property` again with `COMPILE_OPTIONS` option

Reply:
It is not that straight forward, the default flags are not contained in the `COMPILE_OPTION`, I managed to solve this by remove the options by the `STRING` function, then add them back to the `COMPILE_OPTION`, then remove the `COMPILE_OPTION` for specific projects.